ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -652,6 +643,17 @@ export const RuntimePanel = fp.flow(
   const {hasMicroarrayData} = fp.find({cdrVersionId}, cdrVersionListResponse.items) || {hasMicroarrayData: false};
   const [{currentRuntime, pendingRuntime}, setRequestedRuntime] = useCustomRuntime(namespace);
 
+  // if runtime configuration type is a default, override its config with preset values
+  if (!!currentRuntime && currentRuntime.status === RuntimeStatus.Deleted) {
+    if (currentRuntime.configurationType === RuntimeConfigurationType.GeneralAnalysis) {
```

Comment:
nit: this could probably be written slightly more generically, e.g. if we had a mapping of configurationType -> preset, or a helper function to do this lookup.

Reply:
I played around with this a bit but couldn't find an alternative that I was happy with that seemed simpler than the switch case here. Mostly complicated by the fact that the config variable name is different in the two cases (dataprocConfig vs gceConfig).

Reply:
This code doesn't need to know that `dataprocConfig` is null in one case or the other. You can just grab the runtimeTemplate and then set both gceConfig and dataprocConfig according to what's there. Ideally, the code here would be treating these presets opaquely anyways.